sdks/java/io/hcatalog/src/main/java/org/apache/beam/sdk/io/hcatalog/HCatalogIO.java

```diff
@@ -196,11 +246,31 @@ Read withContext(ReaderContext context) {
     }
 
     @Override
+    @SuppressWarnings("deprecation")
     public PCollection<HCatRecord> expand(PBegin input) {
       checkArgument(getTable() != null, "withTable() is required");
       checkArgument(getConfigProperties() != null, "withConfigProperties() is required");
-
-      return input.apply(org.apache.beam.sdk.io.Read.from(new BoundedHCatalogSource(this)));
+      if (!getShouldTreatUnboundedAsBounded()) {
+        checkArgument(getPollingInterval() != null, "withPollingInterval() is required");
+        final Read readSpec =
+            new AutoValue_HCatalogIO_Read.Builder()
+                .setDatabase(DEFAULT_DATABASE)
```

Comment:
Should this really always use DEFAULT_DATABASE?

Reply:
I didn't really need to do this here. Removed this logic of creating a spec from here. However, the current version of HcatalogIO does use `DEFAULT_DATABASE`. I can change it to not do that if we want.